site/using.md

```diff
@@ -7,7 +7,149 @@ All of the features of Flux are accessible from within
 [Weave Cloud](https://cloud.weave.works).
 
 However, `fluxctl` provides an equivalent API that can be used from
-the command line. The `--help` for `fluxctl` is described below.
+the command line.
+
+Download the latest version of the fluxctl client
+[from github](https://github.com/weaveworks/flux/releases).
+
+The `--help` for `fluxctl` is described below.
+
+# Connecting fluxctl to the daemon
+
+You need to tell `fluxctl` where to find the Flux API and to make
+the pod accessible to the command-line client `fluxctl`, you
+will need to expose the API outside the cluster.
+
+A simple way to do this is to piggy-back on `kubectl port-forward`,
+assuming you can access the Kubernetes API:
+
+```sh
+fluxpod=$(kubectl get pod -l name=flux -o name | awk -F / '{ print $2; }')
+kubectl port-forward "$fluxpod" 10080:3030 &
+export FLUX_URL="http://localhost:10080/api/flux"
+```
+
+After that, exporting `FLUX_URL` is enough for `fluxctl` to know how
+to contact the daemon. You could alternatively supply the `--url` argument
+each time.
+
+## Flux API service
+
+Now you can easily query the Flux API:
+
+```sh
+fluxctl list-controllers --all-namespaces
+```
+
+### Local endpoint
+
+**Beware**: this exposes the Flux API, unauthenticated, over an
+insecure channel. Do not do this _unless_ you are operating Flux
+entirely locally; and arguably, only to try it out.
+
+If you are running Flux locally, e.g., in minikube, you can use a
+service with a
+[`NodePort`](http://kubernetes.io/docs/user-guide/services/#type-nodeport).
+
+An example manifest is given in
+[flux-nodeport.yaml](../../deploy/flux-nodeport.yaml).
+
+Then you can access the API on the `NodePort`, by retrieving the port
+number (this example assumes you are using minikube):
+
+```sh
+fluxport=$(kubectl get svc flux --template '{{ index .spec.ports 0 "nodePort" }}')
+export FLUX_URL="http://$(minikube ip):$fluxport/api/flux"
+fluxctl list-controllers --all-namespaces
+```
+
+# Customising the daemon configuration
```

Comment:
This is kind of "advanced configuration", isn't it. What to do what to do ...

Reply:
This one could be just about `fluxctl` -- how to connect it, and what the commands do. Then `setup` could just be about properly setting everything up and how to customise the deployment.